clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -100,7 +100,8 @@ public Errors error() {
         if (responseData == null) {
             synchronized (this) {
                 if (responseData == null) {
-                    responseData = new LinkedHashMap<>();
+                    final LinkedHashMap<TopicPartition, FetchResponseData.PartitionData> responseDataTmp =
```

Comment:
nit: Could we add a comment above this line, to mention we are doing this to avoid other threads accessing a half-initialized object, something like that?

Reply:
Thank you for your advice. Since English is not my mother tongue, I would like to write the sentence "We are doing this to avoid other threads accessing a half-initialized object."

Reply:
How about: `Assigning the lazy-initialized responseData in the last step to avoid other threads accessing a half-initialized object.` ?

Reply:
Ok, thank you. Please review it.